code/datums/components/charger.dm

```diff
@@ -0,0 +1,62 @@
+/**
+  *A component to add to a hostile simplemob to give it a charge attack.
+  **/
+
+/datum/component/charger
+	var/mob/living/simple_animal/hostile/simple_parent
+	///Tracks if the parent is actively charging.
+	var/charge_state = FALSE
+	///In a charge, how many tiles will the parent travel?
+	var/charge_distance = 3
+	///How often can the charging parent actually charge? Effects the cooldown between charges.
+	var/charge_frequency = 6 SECONDS
+	///If the parent is charging, how long will it stun it's target on success, and itself on failure?
+	var/knockdown_time = 3 SECONDS
+	///Declares a cooldown for potential charges right off the bat.
+	COOLDOWN_DECLARE(charge_cooldown)
+
+/datum/component/charger/Initialize(_charge_dist, _charge_freq, _knockdown_time)
+	if(!ishostile(parent))
+		return COMPONENT_INCOMPATIBLE
+	simple_parent = parent
+	if(_charge_dist)
+		charge_distance = _charge_dist
+	if(_charge_freq)
+		charge_frequency = _charge_freq
+	if(_knockdown_time)
+		knockdown_time = _knockdown_time
+	RegisterSignal(parent, list(COMSIG_HOSTILE_CHARGING_TARGET), .proc/enter_charge)
+	RegisterSignal(parent, list(COMSIG_HOSTILE_POST_CHARGE), .proc/charge_check)
+	RegisterSignal(parent, list(COMSIG_HOSTILE_STOP_CHARGE), .proc/charge_end)
+
+/**
+  * Proc that handles a charge attack for a mob.
+  */
+/datum/component/charger/proc/enter_charge(datum/source, var/atom/target)
+	if(!isatom(target))
+		return FALSE
+	var/target_distance = get_dist(simple_parent,target)
+	if((simple_parent.mobility_flags & (MOBILITY_MOVE | MOBILITY_STAND)) != (MOBILITY_MOVE | MOBILITY_STAND) || charge_state)
+		return
+	if((target_distance > simple_parent.minimum_distance) && (target_distance <= charge_distance))
+		return
+	if(!(COOLDOWN_FINISHED(src, charge_cooldown)) || !simple_parent.has_gravity() || !target.has_gravity())
+		return
+	simple_parent.Shake(15, 15, 1 SECONDS)
+	//sleep(1.5 SECONDS) //Provides a visable wind up and tell for all charging mobs, with consistant visuals each time.
```

Comment:
I assume you missed this

Reply:
If this is intentional, please move this to something like `addtimer` so the signal stays non-blocking.